internal/storage/auth/memory/store.go

```diff
@@ -0,0 +1,127 @@
+package memory
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/gofrs/uuid"
+	"go.flipt.io/flipt/errors"
+	"go.flipt.io/flipt/internal/storage"
+	"go.flipt.io/flipt/internal/storage/auth"
+	rpcauth "go.flipt.io/flipt/rpc/flipt/auth"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// Store is an in-memory implementation of storage.AuthenticationStore
+//
+// Authentications are stored in a map by hashedClientToken.
+// Access to the map is protected by a mutex, meaning this is implementation
+// is sage to use concurrently.
```

Comment:
```suggestion
// is safe to use concurrently.
```

although sage sounds cool

Reply:
It is so sage to use concurrently. So very sage.